### ISSUE TITLE
New-TfvcSrcSrvIniContent : Exception calling "GetLocalVersions" with "2" argument(s)

### ISSUE THREAD

**Issue by brockweaver** [Microsoft/vsts-tasks#1907]
We continue to get the following error when the "Publish symbols path" step executes for any build we have configured (changed repo name to "our_repo" for privacy concerns -- real repo name has no underscores):

> 
New-TfvcSrcSrvIniContent : Exception calling "GetLocalVersions" with "2" argument(s): "TF30063: You are not authorized to access https://our_repo.visualstudio.com/."  

All other steps complete successfully and NuGet package is published.  Note the logged user is "buildguest".  No idea where to look next. Please help!

See below for detailed logs.

2016-06-20T17:16:52.6467160Z C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -NoLogo -Sta -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command ". ([scriptblock]::Create('if (!$PSHOME) { $null = Get-Item -LiteralPath ''variable:PSHOME'' } else { Import-Module -Name ([System.IO.Path]::Combine($PSHOME, ''Modules\Microsoft.PowerShell.Management\Microsoft.PowerShell.Management.psd1'')) ; Import-Module -Name ([System.IO.Path]::Combine($PSHOME, ''Modules\Microsoft.PowerShell.Utility\Microsoft.PowerShell.Utility.psd1'')) }')) 2>&1 | ForEach-Object { Write-Verbose $_.Exception.Message -Verbose } ; $env:TF_BUILD = 'True' ; Import-Module -Name 'C:\LR\MMS\Services\Mms\TaskAgentProvisioner\Tools\agents\1.101.1\tasks\PublishSymbols\1.0.27\ps_modules\VstsTaskSdk\VstsTaskSdk.psd1' -ArgumentList @{ NonInteractive = $true } -ErrorAction Stop ; $VerbosePreference = 'Continue' ; $DebugPreference = 'Continue' ; Invoke-VstsTaskScript -ScriptBlock ([scriptblock]::Create('. ''C:\LR\MMS\Services\Mms\TaskAgentProvisioner\Tools\agents\1.101.1\tasks\PublishSymbols\1.0.27\PublishSymbols.ps1'''))"
2016-06-20T17:16:53.8797074Z ##[debug]VstsTaskSdk 0.6.2 commit b4160c33da0010bd83279e9ffa9bbf13bcbd1ce2
2016-06-20T17:16:54.0123012Z ##[debug]Entering C:\LR\MMS\Services\Mms\TaskAgentProvisioner\Tools\agents\1.101.1\tasks\PublishSymbols\1.0.27\PublishSymbols.ps1.
2016-06-20T17:16:54.0287061Z ##[debug]Loading resource strings from: C:\LR\MMS\Services\Mms\TaskAgentProvisioner\Tools\agents\1.101.1\tasks\PublishSymbols\1.0.27\Task.json
2016-06-20T17:16:54.0867064Z ##[debug]Loaded 22 strings.
2016-06-20T17:16:54.1057063Z ##[debug]Env:SYSTEM_CULTURE: 'en-US'
2016-06-20T17:16:54.1057063Z ##[debug]Loading resource strings from: C:\LR\MMS\Services\Mms\TaskAgentProvisioner\Tools\agents\1.101.1\tasks\PublishSymbols\1.0.27\Strings\resources.resjson\en-US\resources.resjson
2016-06-20T17:16:54.2497069Z ##[debug]Loaded 22 strings.
2016-06-20T17:16:54.2617069Z ##[debug]Env:INPUT_SYMBOLSMAXIMUMWAITTIME (empty)
2016-06-20T17:16:54.2687980Z ##[debug] Defaulted to: '0'
2016-06-20T17:16:54.2787082Z ##[debug] Converted to int: 0
2016-06-20T17:16:54.2997084Z ##[debug]Env:INPUT_DELETE (empty)
2016-06-20T17:16:54.3037084Z ##[debug] Converted to bool: False
2016-06-20T17:16:54.3107085Z ##[debug]Env:INPUT_SYMBOLSPATH (empty)
2016-06-20T17:16:54.3227073Z ##[debug]Env:INPUT_SEARCHPATTERN: '**\bin\**\*.pdb'
2016-06-20T17:16:54.3267069Z ##[debug]Env:INPUT_SOURCEFOLDER (empty)
2016-06-20T17:16:54.3317070Z ##[debug]Env:BUILD_DEFINITIONNAME: 'Q2Labs.Common'
2016-06-20T17:16:54.3357153Z ##[debug]Env:INPUT_SYMBOLSPRODUCT (empty)
2016-06-20T17:16:54.3387094Z ##[debug] Defaulted to: 'Q2Labs.Common'
2016-06-20T17:16:54.3437103Z ##[debug]Env:BUILD_BUILDNUMBER: '1.2.36'
2016-06-20T17:16:54.3487086Z ##[debug]Env:INPUT_SYMBOLSVERSION (empty)
2016-06-20T17:16:54.3507148Z ##[debug] Defaulted to: '1.2.36'
2016-06-20T17:16:54.3557093Z ##[debug]Env:BUILD_SOURCESDIRECTORY: 'C:\a\1\s'
2016-06-20T17:16:54.3597092Z ##[debug]Env:INPUT_SYMBOLSFOLDER (empty)
2016-06-20T17:16:54.3627088Z ##[debug] Defaulted to: 'C:\a\1\s'
2016-06-20T17:16:54.3677073Z ##[debug]Env:INPUT_SYMBOLSARTIFACTNAME: 'Symbols_debug'
2016-06-20T17:16:54.3707057Z ##[debug]Env:INPUT_SKIPINDEXING: 'false'
2016-06-20T17:16:54.3737064Z ##[debug] Converted to bool: False
2016-06-20T17:16:54.3787069Z ##[debug]Env:INPUT_TREATNOTINDEXEDASWARNING: 'false'
2016-06-20T17:16:54.3807082Z ##[debug] Converted to bool: False
2016-06-20T17:16:54.4007078Z ##[debug]Entering Find-VstsFiles.
2016-06-20T17:16:54.4157096Z ##[debug] LiteralDirectory: 'C:\a\1\s'
2016-06-20T17:16:54.4187067Z ##[debug] LegacyPattern: '**\bin\**\*.pdb'
2016-06-20T17:16:54.5584925Z ##[debug]Entering Get-MatchingItems.
2016-06-20T17:16:54.5618145Z ##[debug] IncludePatterns: 'C:\a\1\s\**\bin\**\*.pdb'
2016-06-20T17:16:54.5618145Z ##[debug] ExcludePatterns: ''
2016-06-20T17:16:54.5627074Z ##[debug] IncludeFiles: 'True'
2016-06-20T17:16:54.5627074Z ##[debug] IncludeDirectories: 'False'
2016-06-20T17:16:54.5638039Z ##[debug] Force: 'False'
2016-06-20T17:16:55.2047076Z ##[debug]Paths: C:\a\1\s\1\
2016-06-20T17:16:55.2127080Z ##[debug] Q2Labs.Common.Tests\bin\Debug\Q2Labs.Common.pdb
2016-06-20T17:16:55.2127080Z ##[debug] Q2Labs.Common.Tests\bin\Debug\Q2Labs.Common.Tests.pdb
2016-06-20T17:16:55.2137077Z ##[debug] Q2Labs.Common\bin\Debug\Q2Labs.Common.pdb
2016-06-20T17:16:55.2137077Z ##[debug]Leaving Get-MatchingItems.
2016-06-20T17:16:55.2147080Z ##[debug]Total found: 3
2016-06-20T17:16:55.2147080Z ##[debug]Leaving Find-VstsFiles.
2016-06-20T17:16:55.2157085Z Found 3 files.
2016-06-20T17:16:55.2217077Z ##[debug]Loading module from path 'C:\LR\MMS\Services\Mms\TaskAgentProvisioner\Tools\agents\1.101.1\tasks\PublishSymbols\1.0.27\IndexHelpers\IndexHelpers.psm1'.
2016-06-20T17:16:55.2877073Z ##[debug]Adding dbghelp native wrappers.
2016-06-20T17:16:55.3907108Z ##[debug]Exporting function 'Invoke-IndexSources'.
2016-06-20T17:16:55.3937080Z ##[debug]Importing function 'Invoke-IndexSources'.
2016-06-20T17:16:55.4027062Z ##[debug]Entering Invoke-IndexSources.
2016-06-20T17:16:55.4057080Z ##[debug] TreatNotIndexedAsWarning: 'False'
2016-06-20T17:16:55.4127267Z ##[debug]Env:AGENT_HOMEDIRECTORY: 'C:\LR\MMS\Services\Mms\TaskAgentProvisioner\Tools\agents\1.101.1'
2016-06-20T17:16:55.4177078Z ##[debug]Env:AGENT_HOMEDIRECTORY: 'C:\LR\MMS\Services\Mms\TaskAgentProvisioner\Tools\agents\1.101.1'
2016-06-20T17:16:55.4387075Z ##[debug]Asserting leaf path exists: 'C:\LR\MMS\Services\Mms\TaskAgentProvisioner\Tools\agents\1.101.1\Agent\Worker\Tools\Pdbstr\pdbstr.exe'
2016-06-20T17:16:55.4647079Z ##[debug]Entering Add-DbghelpLibrary.
2016-06-20T17:16:55.4727074Z ##[debug]Env:AGENT_HOMEDIRECTORY: 'C:\LR\MMS\Services\Mms\TaskAgentProvisioner\Tools\agents\1.101.1'
2016-06-20T17:16:55.4787072Z ##[debug]Env:AGENT_HOMEDIRECTORY: 'C:\LR\MMS\Services\Mms\TaskAgentProvisioner\Tools\agents\1.101.1'
2016-06-20T17:16:55.4827072Z ##[debug]Asserting leaf path exists: 'C:\LR\MMS\Services\Mms\TaskAgentProvisioner\Tools\agents\1.101.1\Agent\Worker\Tools\Symstore\dbghelp.dll'
2016-06-20T17:16:55.5245933Z ##[debug]Entering Invoke-LoadLibrary.
2016-06-20T17:16:55.5257068Z ##[debug] LiteralPath: 'C:\LR\MMS\Services\Mms\TaskAgentProvisioner\Tools\agents\1.101.1\Agent\Worker\Tools\Symstore\dbghelp.dll'
2016-06-20T17:16:55.5867079Z ##[debug]Leaving Invoke-LoadLibrary.
2016-06-20T17:16:55.5917068Z ##[debug]Leaving Add-DbghelpLibrary.
2016-06-20T17:16:55.6087071Z ##[debug]Entering Get-SourceProvider.
2016-06-20T17:16:55.6147084Z ##[debug]Env:BUILD_REPOSITORY_PROVIDER: 'TfsVersionControl'
2016-06-20T17:16:55.6197070Z ##[debug]Env:BUILD_SOURCESDIRECTORY: 'C:\a\1\s'
2016-06-20T17:16:55.6247080Z ##[debug]Env:SYSTEM_TEAMPROJECTID: 'da726217-cc18-4f1e-8f8d-bd9c5ad6e7da'
2016-06-20T17:16:55.6327091Z ##[debug]Env:AGENT_SERVEROMDIRECTORY: 'C:\LR\MMS\Services\Mms\TaskAgentProvisioner\Tools\agents\1.101.1\agent\worker'
2016-06-20T17:16:56.1712369Z ##[debug]Env:SYSTEM_TEAMFOUNDATIONCOLLECTIONURI: 'https://our_repo.visualstudio.com/'
2016-06-20T17:16:56.1812377Z ##[debug]Env:AGENT_SERVEROMDIRECTORY: 'C:\LR\MMS\Services\Mms\TaskAgentProvisioner\Tools\agents\1.101.1\agent\worker'
2016-06-20T17:16:56.1842369Z ##[debug]Asserting path exists: 'C:\LR\MMS\Services\Mms\TaskAgentProvisioner\Tools\agents\1.101.1\agent\worker\Microsoft.TeamFoundation.Client.dll'
2016-06-20T17:16:56.1932364Z ##[debug]Env:ENDPOINT_URL_SystemVssConnection: 'https://our_repo.visualstudio.com/'
2016-06-20T17:16:56.1972372Z ##[debug]Env:ENDPOINT_AUTH_SystemVssConnection: '********'
2016-06-20T17:16:56.2162364Z ##[debug]Env:ENDPOINT_DATA_SystemVssConnection: '{"ServerId":"2a693934-aba2-4571-a237-207aa47e6cc2","ServerName":"our_repo"}'
2016-06-20T17:16:56.6174334Z ##[debug]Leaving Get-SourceProvider.
2016-06-20T17:16:56.6404309Z ##[debug]Entering Get-SourceFilePaths.
2016-06-20T17:16:56.6424344Z ##[debug] SymbolsFilePath: 'C:\a\1\s\1\Q2Labs.Common.Tests\bin\Debug\Q2Labs.Common.pdb'
2016-06-20T17:16:57.0084346Z ##[debug]Found source files:
2016-06-20T17:16:57.0094303Z ##[debug]Paths: C:\a\1\s\1\Q2Labs.Common\
2016-06-20T17:16:57.0094303Z ##[debug] BaseError.cs
2016-06-20T17:16:57.0104307Z ##[debug] BaseException.cs
2016-06-20T17:16:57.0104307Z ##[debug] BaseJsonException.cs
2016-06-20T17:16:57.0114312Z ##[debug] CacheItemT.cs
2016-06-20T17:16:57.0114312Z ##[debug] CacheT.cs
2016-06-20T17:16:57.0114312Z ##[debug] Data\DataFacility.cs
2016-06-20T17:16:57.0124310Z ##[debug] Data\SafeDataReader.cs
2016-06-20T17:16:57.0124310Z ##[debug] Envelope.cs
2016-06-20T17:16:57.0134302Z ##[debug] Json\JsonSerializer.cs
2016-06-20T17:16:57.0144305Z ##[debug] Json\Utils\ContractResolver.cs
2016-06-20T17:16:57.0164310Z ##[debug] Json\Utils\DateTimeOffsetValueProvider.cs
2016-06-20T17:16:57.0194292Z ##[debug] Json\Utils\DecimalValueProvider.cs
2016-06-20T17:16:57.0224298Z ##[debug] Json\Utils\NullableDateTimeOffsetValueProvider.cs
2016-06-20T17:16:57.0244301Z ##[debug] Json\Utils\NullableDecimalValueProvider.cs
2016-06-20T17:16:57.0274303Z ##[debug] Logging\EventLogWriter.cs
2016-06-20T17:16:57.0304300Z ##[debug] Logging\FileLogWriter.cs
2016-06-20T17:16:57.0334306Z ##[debug] Logging\Logger.cs
2016-06-20T17:16:57.0364303Z ##[debug] Logging\LogItem.cs
2016-06-20T17:16:57.0394305Z ##[debug] Logging\LogWriterBase.cs
2016-06-20T17:16:57.0414311Z ##[debug] Net\BigIntegerExtensions.cs
2016-06-20T17:16:57.0444306Z ##[debug] Net\IPAddressCollection.cs
2016-06-20T17:16:57.0464310Z ##[debug] Net\IPNetwork.cs
2016-06-20T17:16:57.0494304Z ##[debug] Net\IPNetworkCollection.cs
2016-06-20T17:16:57.0524300Z ##[debug] Threading\WorkerBase.cs
2016-06-20T17:16:57.0554292Z ##[debug] Threading\WorkerBaseT.cs
2016-06-20T17:16:57.0584306Z ##[debug] Util.cs
2016-06-20T17:16:57.0614294Z ##[debug]Leaving Get-SourceFilePaths.
2016-06-20T17:16:57.0744296Z ##[debug]Entering New-SrcSrvIniContent.
2016-06-20T17:16:57.0954294Z ##[debug]Entering New-TfvcSrcSrvIniContent.
2016-06-20T17:16:57.6600980Z ##[debug]Leaving New-TfvcSrcSrvIniContent.
2016-06-20T17:16:57.6700993Z ##[debug]Leaving New-SrcSrvIniContent.
2016-06-20T17:16:57.6760988Z ##[debug]Entering Invoke-DisposeSourceProvider.
2016-06-20T17:16:57.6900979Z ##[debug]Leaving Invoke-DisposeSourceProvider.
2016-06-20T17:16:57.6973093Z ##[debug]Entering Remove-DbghelpLibrary.
2016-06-20T17:16:57.7053212Z ##[debug] HModule: '140731886665728'
2016-06-20T17:16:57.7133250Z ##[debug]Leaving Remove-DbghelpLibrary.
2016-06-20T17:16:57.7173188Z ##[debug]Leaving Invoke-IndexSources.
2016-06-20T17:16:57.7209325Z ##[debug]Leaving C:\LR\MMS\Services\Mms\TaskAgentProvisioner\Tools\agents\1.101.1\tasks\PublishSymbols\1.0.27\PublishSymbols.ps1.
2016-06-20T17:16:57.7306341Z ##[debug]Caught exception from task script.
2016-06-20T17:16:57.7366351Z ##[debug]Error record:
2016-06-20T17:16:57.8186357Z ##[debug]New-TfvcSrcSrvIniContent : Exception calling "GetLocalVersions" with "2" argument(s): "TF30063: You are not authorized to access https://our_repo.visualstudio.com/."
2016-06-20T17:16:57.8206346Z ##[debug]At C:\LR\MMS\Services\Mms\TaskAgentProvisioner\Tools\agents\1.101.1\tasks\PublishSymbols\1.0.27\IndexHelpers\SrcSrvIniContentFunctions.ps1:13 char:37
2016-06-20T17:16:57.8216375Z ##[debug]+                 $srcSrvIniContent = New-TfvcSrcSrvIniContent -Provider $Provider ...
2016-06-20T17:16:57.8236356Z ##[debug]+                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2016-06-20T17:16:57.8256350Z ##[debug]    + CategoryInfo          : NotSpecified: (:) [New-TfvcSrcSrvIniContent], MethodInvocationException
2016-06-20T17:16:57.8276359Z ##[debug]    + FullyQualifiedErrorId : TeamFoundationServerUnauthorizedException,New-TfvcSrcSrvIniContent
2016-06-20T17:16:57.8296353Z ##[debug] 
2016-06-20T17:16:57.8326353Z ##[debug]Script stack trace:
2016-06-20T17:16:57.8386350Z ##[debug]at New-TfvcSrcSrvIniContent, C:\LR\MMS\Services\Mms\TaskAgentProvisioner\Tools\agents\1.101.1\tasks\PublishSymbols\1.0.27\IndexHelpers\SrcSrvIniContentFunctions.ps1: line 130
2016-06-20T17:16:57.8396364Z ##[debug]at New-SrcSrvIniContent, C:\LR\MMS\Services\Mms\TaskAgentProvisioner\Tools\agents\1.101.1\tasks\PublishSymbols\1.0.27\IndexHelpers\SrcSrvIniContentFunctions.ps1: line 13
2016-06-20T17:16:57.8416350Z ##[debug]at Invoke-IndexSources, C:\LR\MMS\Services\Mms\TaskAgentProvisioner\Tools\agents\1.101.1\tasks\PublishSymbols\1.0.27\IndexHelpers\IndexFunctions.ps1: line 61
2016-06-20T17:16:57.8436360Z ##[debug]at <ScriptBlock>, C:\LR\MMS\Services\Mms\TaskAgentProvisioner\Tools\agents\1.101.1\tasks\PublishSymbols\1.0.27\PublishSymbols.ps1: line 51
2016-06-20T17:16:57.8446367Z ##[debug]at <ScriptBlock>, <No file>: line 1
2016-06-20T17:16:57.8466350Z ##[debug]at <ScriptBlock>, <No file>: line 22
2016-06-20T17:16:57.8476366Z ##[debug]at <ScriptBlock>, <No file>: line 18
2016-06-20T17:16:57.8496360Z ##[debug]at <ScriptBlock>, <No file>: line 1
2016-06-20T17:16:57.8526343Z ##[debug]Exception:
2016-06-20T17:16:57.8756350Z ##[debug]System.Management.Automation.MethodInvocationException: Exception calling "GetLocalVersions" with "2" argument(s): "TF30063: You are not authorized to access https://our_repo.visualstudio.com/." ---> Microsoft.TeamFoundation.TeamFoundationServerUnauthorizedException: TF30063: You are not authorized to access https://our_repo.visualstudio.com/.
2016-06-20T17:16:57.8776361Z ##[debug]   at Microsoft.TeamFoundation.Client.Channels.TfsHttpWebRequest.EnsureTokenProvider(HttpWebResponse webResponse)
2016-06-20T17:16:57.8786380Z ##[debug]   at Microsoft.TeamFoundation.Client.Channels.TfsHttpWebRequest.SendRequest()
2016-06-20T17:16:57.8806348Z ##[debug]   at Microsoft.TeamFoundation.Client.Channels.TfsHttpRequestChannel.Request(TfsMessage message, TimeSpan timeout)
2016-06-20T17:16:57.8826357Z ##[debug]   at Microsoft.TeamFoundation.Client.Channels.TfsHttpClientBase.Invoke(TfsClientOperation operation, Object[] parameters, TimeSpan timeout, Object[]& outputs)
2016-06-20T17:16:57.8836337Z ##[debug]   at Microsoft.TeamFoundation.VersionControl.Client.Repository.QueryLocalVersions(String workspaceName, String workspaceOwner, ItemSpec[] itemSpecs)
2016-06-20T17:16:57.8856360Z ##[debug]   at Microsoft.TeamFoundation.VersionControl.Client.WebServiceLayerLocalWorkspaces.QueryLocalVersions(String workspaceName, String workspaceOwner, ItemSpec[] itemSpecs)
2016-06-20T17:16:57.8876345Z ##[debug]   at Microsoft.TeamFoundation.VersionControl.Client.Workspace.GetLocalVersions(ItemSpec[] itemSpecs, Boolean sortData)
2016-06-20T17:16:57.8886368Z ##[debug]   at CallSite.Target(Closure , CallSite , Object , Object , Boolean )
2016-06-20T17:16:57.8906364Z ##[debug]   --- End of inner exception stack trace ---
2016-06-20T17:16:57.8926348Z ##[debug]   at System.Management.Automation.ExceptionHandlingOps.CheckActionPreference(FunctionContext funcContext, Exception exception)
2016-06-20T17:16:57.8946348Z ##[debug]   at System.Management.Automation.Interpreter.ActionCallInstruction`2.Run(InterpretedFrame frame)
2016-06-20T17:16:57.8966347Z ##[debug]   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
2016-06-20T17:16:57.8976370Z ##[debug]   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
2016-06-20T17:16:57.9006353Z ##[debug]   at System.Management.Automation.Interpreter.Interpreter.Run(InterpretedFrame frame)
2016-06-20T17:16:57.9016368Z ##[debug]   at System.Management.Automation.Interpreter.LightLambda.RunVoid1[T0](T0 arg0)
2016-06-20T17:16:57.9036338Z ##[debug]   at System.Management.Automation.PSScriptCmdlet.RunClause(Action`1 clause, Object dollarUnderbar, Object inputToProcess)
2016-06-20T17:16:57.9046368Z ##[debug]   at System.Management.Automation.PSScriptCmdlet.DoEndProcessing()
2016-06-20T17:16:57.9066334Z ##[debug]   at System.Management.Automation.CommandProcessorBase.Complete()
2016-06-20T17:16:57.9406344Z ##[error]Exception calling "GetLocalVersions" with "2" argument(s): "TF30063: You are not authorized to access https://our_repo.visualstudio.com/."

**Comment by ericsciple** [comment]
@brockweaver Can you try this again now? I can't repro on the hosted agent, although I have a repro on a specific commit of the extended client SDK.

When you ran into this issue, the hosted agent was at v1.101.1. Since then, a hotfix has rolled out for some other issues and brought it up to v1.101.2. I think it may have also happened to carry a fix for this issue.

**Comment by brockweaver** [comment]
Ran again, same issue.  See attached zip of file logs.

Thank you for your response though, I do appreciate it.

On Tue, Jun 21, 2016 at 2:24 PM, ericsciple <notifications@github.com>
wrote:

> @brockweaver <https://github.com/brockweaver> Can you try this again now?
> I can't repro on the hosted agent, although I have a repro on a specific
> commit of the extended client SDK.
>
> When you ran into this issue, the hosted agent was at v1.101.1. Since
> then, a hotfix has rolled out for some other issues and brought it up to
> v1.101.2. I think it may have also happened to carry a fix for this issue.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Microsoft/vsts-tasks/issues/1907#issuecomment-227544804>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AA_FQ5vbFX5iMGmJyodxlOEKNRVK29qEks5qODpwgaJpZM4I58ER>
> .
>



-- 
Brock Weaver
brockweaver@gmail.com

**Comment by brockweaver** [comment]
Sorry, attachment failed.
[logs_250.zip](https://github.com/Microsoft/vsts-tasks/files/326507/logs_250.zip)

**Comment by ericsciple** [comment]
@brockweaver Can you see if this helps? Before your Publish Symbols task runs, add a PowerShell task that runs this inline script:
```powershell
$f="$env:AGENT_HOMEDIRECTORY\tasks\PublishSymbols\1.0.27\ps_modules\VstsTaskSdk\ServerOMFunctions.ps1"
gc $f|out-string|% {sc $f $_.Replace('    New-Object Microsoft.TeamFoundation.Client.OAuthTokenCredential([string]$endpoint.auth.parameters.AccessToken)', '$c=New-Object Microsoft.TeamFoundation.Client.TfsClientCredentials($false) ; $c.AllowInteractive=$false ; $c.Federated=New-Object Microsoft.TeamFoundation.Client.OAuthTokenCredential([string]$endpoint.auth.parameters.AccessToken) ; $c')}
```

Sorry for the cryptic syntax. I'm working within a max size constraint of the inline script text box. Basically, the goal is to change this line in the Get-TfsClientCredentials function
```powershell
    New-Object Microsoft.TeamFoundation.Client.OAuthTokenCredential([string]$endpoint.auth.parameters.AccessToken)
```
into this line
```powershell
$c=New-Object Microsoft.TeamFoundation.Client.TfsClientCredentials($false) ; $c.AllowInteractive=$false ; $c.Federated=New-Object Microsoft.TeamFoundation.Client.OAuthTokenCredential([string]$endpoint.auth.parameters.AccessToken) ; $c
```

To verify the change took effect, I also added a second inline script task to dump the content of the modified file:
```powershell
dir "$env:AGENT_HOMEDIRECTORY\tasks\PublishSymbols\1.0.27\ps_modules\VstsTaskSdk\ServerOMFunctions.ps1" |gc
```

My repro turned out to be similar only, but not exactly the same issue. In my scenario, I was using server workspaces (new preview agent). I have one other report of someone running into the same issue that you are hitting.

**Comment by brockweaver** [comment]
Bingo!  That worked.  See attached logs for any info you might need.  

Thank you so very much!

[logs_252.zip](https://github.com/Microsoft/vsts-tasks/files/327983/logs_252.zip)

**Comment by ericsciple** [comment]
@brockweaver Thanks for confirmation! The helps me out to figure out a proper fix.

**Comment by ericsciple** [comment]
@brockweaver are you scripting TF.exe during your build (before the publish symbols task runs), or using the TFS extended client DLLs to authenticate to the service? I'm confused why this only repros on certain accounts - trying to rule out whether TFS creds are getting cached locally on the machine and interfering when the publish symbols step runs.

**Comment by brockweaver** [comment]
@ericsciple We used the "Create New Build Definition" template in tfs online.  Whatever steps it includes is what we're doing.  We didn't do any custom scripts or anything.  

It seemed to work fine when we had multiple check out and local workspaces enabled for our repository.  Then we switched our repo to be single check out on all file types and use server workspaces.  I believe that's when the publish symbols path step started failing. I can't say that for sure though -- we changed several settings the day it started to fail.
